packages/util/src/BaseApi.ts

```diff
@@ -174,16 +188,28 @@ export class BaseApi {
     this._historySyncOperations = [...value];
   }
 
-  public getHistory(id: string): History | null {
-    if (!id) return null;
+  public getHistory(id: string): HistoryItem | null {
+    return this.history[id] ?? null;
+  }
+
+  public getFilteredHistory(filterFn: (item: HistoryItem) => boolean): AccountHistory<HistoryItem> {
+    const current = this.history;
+    const filtered: AccountHistory<HistoryItem> = {};
 
-    return this.history.find((item) => item.id === id) ?? null;
+    for (const id in current) {
```

Comment:
I would name **currentHistory**, like so it is easier to be read without going back by lines to figure out